ports/geographiclib/portfile.cmake

```
@@ -1,49 +1,51 @@
-vcpkg_from_sourceforge (
+vcpkg_from_sourceforge(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO geographiclib
-    REF distrib
-    FILENAME "GeographicLib-1.52.tar.gz"
-    SHA512 98a4d33764db4a4755851a7db639fd9e055dcf4f1f949258e112fce2e198076b5896fcae2c1ea36b37fe1000d28eec326636a730e70f25bc19a1610423ba6859
-    PATCHES cxx-library-only.patch
+    REF distrib-C++
+    FILENAME "GeographicLib-2.0.tar.gz"
+    SHA512 7cf67174a64082372cdd249a64460e9f61c582aaf3d2a31e4e69d811f265e078ba62f945e9f1f44be6c58de4c20d0359dd46e0fd262ffac229df0ba2c6adc848
+    )
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES "tools" TOOLS
 )
 
-vcpkg_check_features (
-  OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-  INVERTED_FEATURES
-    "tools" SKIP_TOOLS
-)
-
-if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-  set (LIB_TYPE "SHARED")
-else ()
-  set (LIB_TYPE "STATIC")
-endif ()
+if(TOOLS)
+    set(TOOL_OPTION "-DBINDIR=tools/${PORT}")
+else()
+    set(TOOL_OPTION -DBINDIR=)
+endif()
 
-vcpkg_configure_cmake (
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    OPTIONS
-        "-DGEOGRAPHICLIB_LIB_TYPE=${LIB_TYPE}"
-        ${FEATURE_OPTIONS}
-    PREFER_NINJA # Disable this option if project cannot be built with Ninja
+    OPTIONS ${LIB_OPTION} ${TOOL_OPTION}
+    "-DCMAKEDIR=share/${PORT}"
+    -DPKGDIR=
+    -DDOCDIR=
+    -DEXAMPLEDIR=
+    -DMANDIR=
+    -DSBINDIR=
```

```suggestion
    OPTIONS
        "-DBINDIR=tools/${PORT}"
        "-DCMAKEDIR=share/${PORT}"
        "-DDOCDIR=share/${PORT}/doc"
        "-DEXAMPLEDIR=share/${PORT}/doc"
        "-DMANDIR=share/${PORT}/man"
        "-DSBINDIR=share/${PORT}/tools"
```

`LIB_OPTION` is not defined.
`BINDIR` should always be set to _tools/${PORT}_.
We should keep `PKGDIR` default value and call `vcpkg_fixup_pkgconfig()` after call `vcpkg_cmake_install`.

---

I defined LIB_OPTION (a mistake on my part that it wasn't set).
I keep PKGDIR set so the pc file is installed.
I didn't set BINDIR to tools/${PORT} because in a previous submission I was told that tools should not be installed unless the tools option is given.  Also the original port (not by me) skipped installing the man pages, doc, etc.  Where should I look for the vcpkg standards on this?

---

Tools should use option `TOOLS` to decide whether to enable or not, not based on this path.

---

> Tools should use option `TOOLS` to decide whether to enable or not, not based on this path.

I would have chosen a different wording. But I do share the concern:
- I would interprete empty paths as a request to use the default - similar to unset variables.
- I would prefer a clear variable names to control options (`BUILD_TOOLS`, `ENABLE_TOOLS` or `WITH_TOOLS`).

Of course, it is upstream to decide whether to change it in a future release. For 2.0, vcpkg has to use the interface which is provided. So this part of the PR is okay: Setting an empty path means turning the feature off. Maybe this should be written into a comment, because it is unusual enough to surprise future readers.

---

In order to deal with the varying needs of multiple packagers, I decided that specifying an install path for each component with an empty path meaning "don't install this component" was the most flexible approach. This let me dispense with a 142-line patch file and replace it with half a dozen install directories set to null.

The logic in CMakeLists does not distinguish `-DBINDIR=OFF` and `-DBINDIR=`, so I've changed the portfile to use `OFF` to mean don't install.  Maybe that will reduce the confusion.  I've also added a comment to explain the behavior